File: GitToolBox/src/main/java/zielu/gittoolbox/ui/projectview/DecorationColorsTextAttributesUi.java
package zielu.gittoolbox.ui.projectview;

import com.intellij.openapi.editor.colors.TextAttributesKey;
import com.intellij.ui.SimpleTextAttributes;
import org.jetbrains.annotations.NotNull;
import zielu.gittoolbox.config.DecorationColors;

public class DecorationColorsTextAttributesUi implements TextAttributesUi {
  private static final TextAttributesUi INSTANCE = new DecorationColorsTextAttributesUi();

  private DecorationColorsTextAttributesUi() {
  }

  public static TextAttributesUi getInstance() {
    return INSTANCE;
  }

  @Override
  public SimpleTextAttributes getTextAttributes(@NotNull TextAttributesKey key) {
    return DecorationColors.simpleAttributes(key);
  }
}

File: src/main/kotlin/zielu/gittoolbox/ui/projectview/DecorationColorsTextAttributesUi.kt
package zielu.gittoolbox.ui.projectview

import com.intellij.openapi.editor.colors.TextAttributesKey
import com.intellij.ui.SimpleTextAttributes
import zielu.gittoolbox.config.DecorationColors

internal object DecorationColorsTextAttributesUi : TextAttributesUi {

  @JvmStatic
  fun getInstance(): TextAttributesUi {
    return this
  }

  override fun getTextAttributes(key: TextAttributesKey): SimpleTextAttributes {
    return DecorationColors.simpleAttributes(key)
  }
}
